src/jit/lowerxarch.cpp

```diff
@@ -2571,39 +2572,56 @@ void Lowering::TreeNodeInfoInitSIMD(GenTree* tree)
                 info->srcCount = 1;
             }
 
-            // If the index is not a constant, we will use the SIMD temp location to store the vector.
-            // Otherwise, if the baseType is floating point, the targetReg will be a xmm reg and we
-            // can use that in the process of extracting the element.
-            //
-            // If the index is a constant and base type is a small int we can use pextrw, but on AVX
-            // we will need a temp if are indexing into the upper half of the AVX register.
-            // In all other cases with constant index, we need a temp xmm register to extract the
-            // element if index is other than zero.
-
-            if (!op2->IsCnsIntOrI())
+            GenTree* op1 = tree->gtOp.gtOp1;
```

Comment:
Suggestion: It's a bit weird that this method declares an `op2` but not an `op1` above. I would suggest making it more consistent - either by having all the cases declare their own variables if they need them (of course that's a bigger impact) or simply declaring both `op1` and `op2` above.